activesupport/lib/active_support/deprecation/reporting.rb

```diff
@@ -37,6 +41,44 @@ def silence(&block)
         @silenced_thread.bind(true, &block)
       end
 
+      # Allow previously disallowd deprecation warnings within the block.
+      # <tt>allowed_warnings<tt> can be an array containing strings, symbols, or regular
+      # expressions. (Symbols are treated as strings). These are compared against
+      # the text of deprecation warning messages generated within the block.
+      # Matching warnings will be exempt from the rules set by
+      # +ActiveSupport::Deprecation.disallowed_warnings+
+      #
+      # The optional <tt>if:</tt> argument accepts a truthy/falsy value or an object that
+      # responds to <tt>.call</tt>. If truthy, then matching warnings will be allowed.
+      # If falsey then the method yields to the block without allowing the warning.
+      #
+      #   ActiveSupport::Deprecation.disallowed_behavior = :raise
+      #   ActiveSupport::Deprecation.disallowed_warnings = [
+      #     "something broke"
+      #   ]
+      #
+      #   ActiveSupport::Deprecation.warn('something broke!')
+      #   # => ActiveSupport::DeprecationException
+      #
+      #   ActiveSupport::Deprecation.allow ['something broke'] do
+      #     ActiveSupport::Deprecation.warn('something broke!')
+      #   end
+      #   # => nil
+      #
+      #   ActiveSupport::Deprecation.allow ['something broke'], if: Rails.env.production? do
+      #     ActiveSupport::Deprecation.warn('something broke!')
+      #   end
+      #   # => ActiveSupport::DeprecationException for dev/test, nil for production
+      def allow(allowed_warnings = :all, if: true, &block)
+        if_conditional = binding.local_variable_get(:if)
+        if_conditional = if_conditional.call if if_conditional.respond_to?(:call)
+        if if_conditional
```

Comment:
Would renaming `if_conditional` to `conditional` make this method body easier to read?

Reply:
I'm not opposed to this. Changed in https://github.com/rails/rails/pull/37940/commits/df0959bd3e4a852d0d4ac7569227be5ada16c251